ftw/linkchecker/command/broken_link.py

```diff
@@ -0,0 +1,73 @@
+from plone import api
+from plone.dexterity.utils import safe_utf8
+
+
+class BrokenLink(object):
+    table_attrs = [
+        'is_internal',
+        'link_origin',
+        'link_target',
+        'status_code',
+        'content_type',
+        'response_time',
+        'header_location',
+        'error_message',
+    ]
+    counter = 0
+
+    def __init__(self):
+        self.is_broken = None
+        self.is_internal = None
+        self.link_origin = 'Unknown origin'
+        self.link_target = 'Unknown target'
+        self.status_code = 'Unknown status code'
+        self.content_type = 'Unknown content type'
+        self.response_time = 'Unknown response time'
+        self.header_location = 'Unknown header location'
+        self.error_message = 'No error occurred'
+        self.counter = BrokenLink.counter
+        BrokenLink.counter += 1
+
+    def __iter__(self):
+        for attr in self.table_attrs:
+            value = getattr(self, attr, '')
+            yield value if isinstance(value, basestring) else ''
+
+    def __len__(self):
```

Comment:
Is this used somewhere?
I'm not sure what I expect when I do:

```
obj = BrokenLink()
print len(obj)
>>> 8
```

O think this is kinda misleading and using `len(obj.table_attrs)` would be totally fine and more readable.

Reply:
Yes it is. But since the naming might be misleading I remove the length function from the class and use .table_attrs instead where I was using the length.